mods/pokebattler.php

```diff
@@ -109,96 +109,55 @@
         $raidlevels[] = 'RAID_LEVEL_' . $level;
     }
     debug_log($raidlevels);
+    $levels_processed = [];
+    $bosses = [];
+    // Process breaking news section
+    $now = new DateTime('now', new DateTimeZone($config->TIMEZONE));
+    $ph = new dateTimeZone('America/Phoenix');
+    foreach($pb_data['breakingNews'] as $news) {
+        if($news['type'] == 'RAID_TYPE_RAID') {
+            $rl = str_replace('RAID_LEVEL_','', $news['tier']);
+            if($rl == "MEGA") $raid_level_id = 6; else $raid_level_id = $rl;
+            $starttime = new DateTime("@".substr($news['startDate'],0,10));
+            $endtime = new DateTime("@".substr($news['endDate'],0,10));
+            $starttime->setTimezone($ph);
+            $endtime->setTimezone($ph);
+
+            if(in_array($news['tier'], $raidlevels) && $starttime->getTimestamp() < $now->getTimestamp() && $endtime->getTimestamp() > $now->getTimestamp()) {
+                $levels_processed[$raid_level_id] = $news['tier'];
+                $dex_id_form = resolve_boss_name_to_ids($news['pokemon']);
+                $bosses[$raid_level_id][] = $dex_id_form;
+            }
+        }
+    }
 
     // Process raid tier(s)
     debug_log('Processing received pokebattler raid bosses for each raid level');
-    // Init index to process the json
-    $index = 0;
     foreach($pb_data['tiers'] as $tier) {
+        $rl = str_replace('RAID_LEVEL_','', $tier['tier']);
+        if($rl == "MEGA") $raid_level_id = 6; else $raid_level_id = $rl;
         // Process raid level?
-        if(!in_array($tier['tier'],$raidlevels)) {
-            // Index + 1
-            $index = $index + 1;
+        if(!in_array($tier['tier'], $raidlevels) or isset($levels_processed[$raid_level_id])) {
```

Comment:
Add a comment here that we're doing this to skip anything that came from breakingNews.